app/helpers/application_helper.rb

```diff
@@ -56,4 +56,62 @@ def time_tag(date_or_time, *args, &block)
   def time_tag_with_title(date_or_time, &block)
     time_tag(date_or_time, title: l(date_or_time, format: :long), &block)
   end
+
+  # used to turn references to threads into hyperlinks
+  #
+  # examples for threads:
+  #  thread 1051
+  #  thread no 1051
+  #  thread no. 1051
+  #  thread number 1051
+  #  thread #1051
+  #  #t1051
+  # examples for issues:
+  #  issue 1051
+  #  issue no 1051
+  #  issue no. 1051
+  #  issue number 1051
+  #  issue #1051
+  #  #i1051
+  def message_linkify(message)
+    THREAD_FORMAT_MAP.each do |key, value|
+      threads_found = message.scan(value)
+
+      threads_found.each do |t|
+        thread_id = t.match(/\d+/)[0]
+        message.gsub!(t, "<a href=\"#{thread_path(thread_id.to_i)}\">#{t}</a>") if thread_id
+      end
+    end
+
+    ISSUE_FORMAT_MAP.each do |key, value|
+      issues_found = message.scan(value)
+
+      issues_found.each do |i|
+        issue_id = i.match(/\d+/)[0]
+        # not a fan of making a database call here. Not sure how else to address parameterizing the issue url.
```

Comment:
Ahh, I see this called on each view, I thought it was called on save...
If @mvl22 is happy the slow down is minor.